CONFIGURATION.md

```diff
@@ -61,6 +62,30 @@ require('instana-nodejs-sensor')({
 });
 ```
 
+## Reporting Uncaught Exceptions
+The Instana Node.js sensor has the ability to report uncaught exceptions. By default, a Node.js process will be terminated by an uncaught exception (see [Node.js docs](https://nodejs.org/api/process.html#process_event_uncaughtexception)). If uncaught exception reporting is enabled, the Instana Node.js sensor will register a listener for the `uncaughtException` event and take the following actions when an uncaught exception occurs:
+
+* Report this as an incident to Instana, including the uncaught exception and its stack trace.
+* Finish the currently active span and mark it as an error (if automatic tracing is enabled).
+
+It will then rethrow the original exception to terminate the Node.js process. (Keeping a Node.js process running after an uncaught exception has occurred is strongly discouraged, as the process might be in an inconsistent state, see [Node.js docs](https://nodejs.org/api/process.html#process_warning_using_uncaughtexception_correctly).)
+
+Reporting uncaught exceptions is disabled by default. It can be enabled with the option `reportUncaughtException`, as follows:
+
+```javascript
+require('instana-nodejs-sensor')({
+  reportUncaughtException: true
+});
+```
+
+The [Node.js docs](https://nodejs.org/api/process.html#process_event_uncaughtexception) advise that the "correct use of 'uncaughtException' is to perform synchronous cleanup of allocated resources (e.g. file descriptors, handles, etc) before shutting down the process." Instana adheres to this and only executes synchronous actions before rethrowing the error.
+
+The downside of this is that it might prolong the time it takes for the application process to finally terminate after the uncaught exception. Since the actions are synchronous, no other incoming requests will be accepted by your application during this time, that is, the process will be unresponsive to incoming request (HTTP, WebSockets, ...). This is an important safeguard, since the application might already be in an inconsistent state, so it would not be safe to accept/process any more requests in this process. However, this has two drawbacks:
+
+* If you have a mechanism in place that restarts the Node.js process once it has crashed, enabling uncaught exception tracking might lead to more failed requests, because the application process is kept alive for a bit longer, delaying the automatic restart.
```

Comment:
It would be great if we could give some indication for how long the process will be kept alive. At least this is some information I would want to have when making this decision.

Reply:
A few milliseconds. Basically as long as it needs to open a new TCP connection to the agent (or fail doing so) and write the required data into the socket buffer. I'll add something there.